[Kwok Ying Kwan Kasey] iP

kaseykwok opened the pull request.

kaseykwok committed:
Complete Level-0's Greeting

kaseykwok committed:
Change chatbot name to Ever

kaseykwok committed:
Add echos on user input and exit at 'bye' input

kaseykwok committed:
Add methods for add and list items

kaseykwok committed:
Add class Task, methods for changing task status

kaseykwok committed:
Correct naming for constants

okkhoy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,63 @@
+            "|  __|   \\ \\/ / |  __| |  _  / \n" +
+            "| |____   \\  /  | |____| | \\ \\ \n" +
+            "|______|   \\/   |______|_|  \\_\\";
+    private static Task[] tasks = new Task[100];
```
I can see a magic number here. Please try to eliminate magic literals

okkhoy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,63 @@
+        String MESSAGE = "Hello! I'm Ever\n" +
+                "What can I do for you?";
+        System.out.println(MESSAGE);
+    }
+    private static void exit() {
```
Whitespace between methods please

okkhoy reviewed (commented):
Please take a look at some comments on your code.

kaseykwok committed:
Add todo class and help command

kaseykwok committed:
Add Deadline class and allow add deadline task

kaseykwok committed:
Add Event class and event command

kaseykwok committed:
Refactor constants, add condition checking for add and list task

dhanish265 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+            } else if (inputWords[0].equals("mark")) {
+                int taskIndex = Integer.parseInt(inputWords[1]) - 1;
+                tasks[taskIndex].markAsDone();
+                System.out.printf("Marked as done: %s\n", tasks[taskIndex].getPrintString());
```
you can consider implementing SLAP to segregate lower and higher level functions.

dhanish265 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,29 @@
+        isDone = false;
+    }
+
+    public String getPrintString() {
```
can consider overriding Object's toString method as that is the intended effect of this method and makes it easier to use this method in other classes

dhanish265 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+        else {
+            System.out.println("Maximum number of tasks reached");
+        }
+    }
```
can consider segregating task related methods to a different class, such as TaskManager

dhanish265 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+    private static int taskCount = 0;
+
+    private static void greet() {
+        String MESSAGE = "Hello! I'm Ever\n" +
```
you can consider making this a final string so that there will not be multiple creations of the same string (although I understand the function is only called once)

kaseykwok committed:
Refactor task adding functions for todo, deadline and event

kaseykwok committed:
Add Task Manager class

kaseykwok committed:
Add test for automated text UI testing

kaseykwok committed:
Add error handling for mark and unmark

kaseykwok committed:
Add Message class and error handling for adding tasks

kaseykwok committed:
Add Command class and give command syntax in error message

kaseykwok committed:
Merge branch 'branch-Level-5'

kaseykwok committed:
Add illegal task number and task count exceed max exceptions

kaseykwok committed:
Merge branch 'branch-A-Exceptions'

kaseykwok committed:
Organise classes into package

kaseykwok committed:
Merge branch 'branch-A-Packages'

kaseykwok committed:
Store commands in ListedHashMap instead of Map

kaseykwok committed:
Use ArrayList to store tasks and add delete task function

kaseykwok committed:
Add save tasks and FileHandler class

kaseykwok committed:
Merge branch 'branch-Level-6'

kaseykwok committed:
Return on save when there are no tasks

kaseykwok committed:
Merge branch 'branch-Level-7'

kaseykwok committed:
Minor fix in save format

kaseykwok committed:
Amend test cases

kaseykwok committed:
Merge branch 'branch-Level-7'

kaseykwok committed:
Amend test result for duke.txt

kaseykwok committed:
Add META file

kaseykwok committed:
Auto save and load duke.txt

kaseykwok committed:
Merge branch 'branch-Level-7'

kaseykwok committed:
Remove save success message and save condition checking

sevenseasofbri commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,109 @@
+        String command = input.split(" ", 2)[0];
+
+        switch (command) {
+        case "help":
```
Consider refactoring the magic literals into `final` variables and then utilise those in your code.

sevenseasofbri commented on src/main/java/duke/Message.java:
```diff
@@ -0,0 +1,34 @@
+    public static final String MAXIMUM_TASKS_REACHED_ERROR_MESSAGE = "Maximum number of tasks reached";
+
+    /** Task Number Error Messages **/
+    public static final String WRONG_TASK_NUMBER_RANGE_ERROR_MESSAGE = "Sorry, the task number is out of range. Type \"list\" for viewing all the tasks.";
+    public static final String WRONG_TASK_NUMBER_ERROR_MESSAGE = "Sorry, the selected task has not been created yet. Type \"list\" to see the task numbers.";
```
Some of these lines are quite long (>120 char). Maybe make them multi-line or shorten them

sevenseasofbri commented on src/main/java/duke/task/TaskManager.java:
```diff
@@ -0,0 +1,196 @@
+package duke.task;
+
+import duke.*;
```
Avoid using wildcard imports. Consider importing each class you require from the package.

sevenseasofbri commented on src/main/java/duke/task/TaskManager.java:
```diff
@@ -0,0 +1,196 @@
+    private String[] parseFileLine(String line) {
+        return line.split(FILE_STRING_SEPARATOR);
+    }
+    public void loadTasks() {
```
Perhaps keep at least 1 line space between methods

sevenseasofbri reviewed (commented):
Hi Kasey, good job on the iP so far! I have added some comments on the code quality, coding standard violations and naming conventions. Hope they will be helpful. 👍🏽

kaseykwok committed:
Remove wild card imports and fix code spacing

kaseykwok committed:
Rename TaskManager to TaskList and task list variable name

kaseykwok committed:
Update test expected files and bash file

kaseykwok committed:
Add Parser and Storage class, more OOP

- Add Parser class to parse data file
- Add Storage class to load and save tasks
- Replace magic literals with constants for Task and its subclasses
- Add constructor that overload with task status for Task classes

kaseykwok committed:
Add parse function for commands and command related classes

- Refactor command parsing in Duke class to parseCommand function in Parser class
- Add Command, CommandDocumentation and CommandMenu class
- Change parameters for commands that input taskNumber in syntax
- Change the compile line for test bash file

kaseykwok committed:
Remove extract parameter method in Task and its subclass

kaseykwok committed:
Add Ui Class, refactor ui methods and clean unused functions

- Add Ui class
- Move all print line functions to Ui class
- Remove unused functions and packages imported
- Remove IllegalTaskNumberInputException
- Change in test bash file

kaseykwok committed:
Code cleanup

kaseykwok committed:
Throw exception if error occur reading file

kaseykwok committed:
Add JavaDoc for class Task, Deadline, Event and Todo

kaseykwok committed:
Add JavaDoc for Storage class

kaseykwok committed:
More details for Storage class's JavaDoc

kaseykwok committed:
Add JavaDoc for Parser class

kaseykwok committed:
Add javaDoc for Ui class

kaseykwok committed:
Add JavaDoc for CommandDocumentation and CommandMenu class

kaseykwok committed:
Add JavaDoc for Command class, clean code for its constructor

kaseykwok committed:
Add JavaDoc for Duke and TaskList class

kaseykwok committed:
Add JavaDoc for Message class

kaseykwok committed:
Code cleaning in single-lined comment

kaseykwok committed:
Add Find operation in task list and change the testing bash file

- Add find operation in task list
- Change the testing bash file
- Add constructor for temporary task list

kaseykwok committed:
Merge pull request #1 from kaseykwok/branch-Level-9

kaseykwok committed:
Merge branch 'master'

kaseykwok committed:
Add JavaDoc for the find operation related methods

kaseykwok committed:
Fix bug in dealing with invalid commands

kaseykwok committed:
Merge pull request #2 from kaseykwok/branch-A-JavaDoc

kaseykwok committed:
Update user guide with some features completed

kaseykwok committed:
Fix table of contents links in user guide

kaseykwok committed:
Complete features section in user guide

kaseykwok committed:
Complete user guide

kaseykwok committed:
Fix possible rendering issue in user guide

kaseykwok committed:
Add a blank line before code blocks in user guide
